common/lib/xmodule/xmodule/contentstore/mongo.py

```diff
@@ -501,6 +501,14 @@ def ensure_indexes(self):
             sparse=True,
             background=True
         )
+        create_collection_index(
+            self.fs_files,
+            [
+                ('displayname', pymongo.TEXT)
+            ],
+            sparse=True,
+            background=True
+        )
```

Comment:
@edx/devops I'm a little concerned about how much space this will take up and how much and how it will affect mongo performance when the index is created. [The mongo docs have a list of some considerations about text indices](https://docs.mongodb.com/manual/core/index-text/#storage-requirements-and-performance-costs). How many files are in the contentstore on prod? Do we have enough extra space in mongo for a index containing every filename?